src/Compilers/CSharp/Portable/Symbols/Source/SourceParameterSymbol.cs

```diff
@@ -251,17 +252,27 @@ public sealed override ImmutableArray<Location> Locations
         {
             get
             {
-                return _locations;
+                return _locations.ToImmutable();
```

Comment:
I am curious if, in the One case, this method should actually cache the array it creates on the containing OneOrMany. i.e. "move" the value from the One field to the Many field. Perhaps there is a synchronization burden which makes it not worth it.

Reply:
i wasn't seeing the allocations show up as an issue after this *for this type*, so i'm wary about caching.  i will be looking for this if/when i may update certain other types (FieldSymbols are also showing up high here for example), to see if we're just shuffling costs around.